parser/src/main/parsing/statements/Stmt.c

```diff
@@ -30,16 +31,27 @@ struct Stmt* makeStmt(struct TokenList* tokens, bool debug) {
 	res->m3 = NULL;
 	res->m4 = NULL;
 	res->m5 = NULL;
+	res->m6 = NULL;
 
 	struct TokenList* copy = list_copy(tokens);
 
 	struct Token* first = list_head(copy);
 
-	if (first->kind == LOOP) {
+	if (first->kind == BREAK){
+		res->m6 = makeBreakStmt(copy, debug);
+	
+		if(res->m6 == NULL){
+			printf("expected break stmt, but was:\n");
+			list_print(copy);
+			freeTokenListShallow(copy);
+			free(res);
+			exit(1);
+		}
```

Comment:
At line 124: res->m1 not free'd.

Reply:
2c1d6bda81685e0989c50766a5f905c4da254e7c